temporalcli/commands.server.go

```diff
@@ -60,6 +60,10 @@ func (t *TemporalServerStartDevCommand) run(cctx *CommandContext, args []string)
 			_, _ = cctx.Options.Stderr.Write(b)
 		}
 	}
+	// Grab a free port for metrics ahead-of-time so we know what port is selected
+	if opts.MetricsPort == 0 {
+		opts.MetricsPort = devserver.MustGetFreePort()
+	}
```

Comment:
Hrmm, so looking at the dev server code, I am not seeing that metrics config is even set if no metrics port is given by the user which I wonder if that means that no metrics endpoint is even created/bound. Can you confirm what happens today if metrics port is unset (i.e. 0)?

Reply:
It is indeed created and started still - but only with LiteServer (which is the one that gets used here, right?) https://github.com/temporalio/temporal/blob/a2b17bebadc914f6445e9bd85be126f94730b25f/internal/temporalite/lite_server.go#L121

Reply:
> which is the one that gets used here, right?

I don't think so. But I wonder if I introduced a regression copying over dev server code. Current CLI does default a metrics port: https://github.com/temporalio/cli/blob/7267df668277a9b43c50d454af50ff5e2b9f8e90/server/config/config.go#L153-L155. But new CLI does not. I kinda like the behavior of not starting/binding a metrics server if they don't ask for it now that I think about it. But I think your current code re-institutes what was in the CLI before.

Can be convinced either way (can also get others' opinion on whether we should always start a metrics server if you want).

Reply:
I figure we might as well. I'm just not sure I see any real reason not too, and it's nice to have the metrics there if at some point after starting it you realize you wanted to see something.

Reply:
Eh, I think the discoverability is a nice thing actually. A curious person might open it up, go "oh hey this is nice" who otherwise never would've known.